core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -509,6 +512,11 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
 
 private[spark] object SparkConf extends Logging {
 
+  // Deprecation message for memory fraction configs used in the old memory management model
+  private val deprecatedMemoryFractionMessage =
+    "As of Spark 1.6, execution and storage memory management are unified. " +
+      "All memory fractions used in the old model are now deprecated and no longer read."
```

Comment:
The old configurations will still be respected in legacy mode, so this is slightly ambiguous / confusing. Is there an easy way to avoid the warning if the legacy mode configuration is turned on? If not, I suppose we could just expand the deprecation message to mention this corner case, perhaps by just appending a "(unless spark.XX.YY is enabled)" at the end.